lib/entities/agent-run.ts

```diff
@@ -0,0 +1,66 @@
+import { freezeSys, toPlainObject } from 'contentful-sdk-core'
+import copy from 'fast-copy'
+import type { DefaultElements, Link, MakeRequest, MetaSysProps } from '../common-types'
+import { wrapCollection } from '../common-utils'
+
+export type AgentRunStatus = 'IN_PROGRESS' | 'FAILED' | 'COMPLETED' | 'PENDING_REVIEW' | 'DRAFT'
+
+export type AgentRunMessageRole = 'system' | 'user' | 'assistant' | 'tool'
+
+export type AgentRunMessageType = 'text' | 'tool-call' | 'tool-result'
+
+export type AgentRunMessageTextPart = {
+  type: 'text'
+  text: string
+}
+
+export type AgentRunMessageToolCallPart = {
+  type: 'tool-call'
+  toolCallId: string
+  toolName: string
+  args: unknown
+}
+
+export type AgentRunMessagePart = AgentRunMessageTextPart | AgentRunMessageToolCallPart
+
+export type AgentRunMessage = {
+  id: string
+  createdAt: string
+  role: AgentRunMessageRole
+  type?: AgentRunMessageType
+  content: {
+    parts: Array<AgentRunMessagePart>
+  }
+}
+
+export type AgentRunProps = {
```

Comment:
[nit] I'm bit confused about the naming, this is a response-type, maybe call it AgentRunResponse instead of a Prop, as that is commonly associated with input types